### PR TITLE
Adding different lifecycle states, creating roles and fixing port offset issue of sample populator

### PR DESCRIPTION
**Pull request by Rajith90**
This PR will create Publisher and Store user roles,
add different life cycle states,
and fixes port offset issue of the sample populator

### CODE REVIEW

File: modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/CategorizeArtifacts.java [hunk 1]
```diff
@@ -11,29 +11,39 @@
...
+* This class add categories,tags to rest and soap services, and create publisher and
+* store user roles.
+ */
 public class CategorizeArtifacts {
+    private static String cookie;
     private static ConfigurationContext configContext = null;
-
+    private static LifeCycleManagementClient lifeCycleManagementClient;
     private static final String CARBON_HOME = System.getProperty("carbon.home");
     private static final String axis2Repo = CARBON_HOME + File.separator + "repository" + File.separator + "deployment" + File.separator + "client";
```
Comment (sameerak) [inline modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/CategorizeArtifacts.java]: code line 2 long.
Please consider dividing in to 2 lines

Comment (Rajith90) [inline modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/CategorizeArtifacts.java]: Refer the commit
https://github.com/Rajith90/product-greg/commit/0d2c0d6bbe023fa08a8942fd6dd7a4f2332dc117

File: modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/Main.java [hunk 2]
```diff
@@ -117,36 +141,94 @@ public void setCookie(String cookie) {
...
+     *This method is used to upload sample WADLs using gar file.
+     *
+     * @param resourceServiceClient
+     * @param projectPath  absolute path of the gar file.
+     * @throws Exception
+     */
     private static void addWadlGar(ResourceServiceClient resourceServiceClient, String projectPath) throws Exception {
 	DataHandler dh = new DataHandler(new URL("file://"+projectPath+"/resources/wadls.gar"));
         resourceServiceClient.addResource("/_system/governance/trunk/test/wadls/",
                 "application/vnd.wso2.governance-archive", null, dh, null, null);
```
Comment (Rajith90) [inline modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/Main.java]: "application/vnd.wso2.governance-archive" Moved to a final string 
#367

File: modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/Main.java [hunk 3]
```diff
@@ -117,36 +141,94 @@ public void setCookie(String cookie) {
...
+     *
+     * @param resourceServiceClient
+     * @param projectPath  absolute path of the gar file.
+     * @throws Exception
+     */
 	private static void addWsdlGar(ResourceServiceClient resourceServiceClient, String projectPath) throws Exception {       
         DataHandler dh = new DataHandler(new URL("file://"+projectPath+"/resources/wsdl_new.gar"));
         resourceServiceClient.addResource("/_system/governance/trunk/test/wsdls/",
-                                               "application/vnd.wso2.governance-archive", null, dh, null,null);
+                "application/vnd.wso2.governance-archive", null, dh, null, null);
```
Comment (Rajith90) [inline modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/Main.java]: "application/vnd.wso2.governance-archive" Moved to a final string 
#367

File: modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/Main.java [hunk 4]
```diff
@@ -117,36 +141,94 @@ public void setCookie(String cookie) {
...
+     *This method is used to upload sample schemas using gar file.
+     *
+     * @param resourceServiceClient
+     * @param projectPath  absolute path of the gar file.
+     * @throws Exception
+     */
 	private static void addSchemaGar(ResourceServiceClient resourceServiceClient, String projectPath) throws Exception {
 	DataHandler dh = new DataHandler(new URL("file://"+projectPath+"/resources/schemas.gar"));
         resourceServiceClient.addResource("/_system/governance/trunk/test/schemas/",
                                                "application/vnd.wso2.governance-archive", null, dh, null,null);
```
Comment (Rajith90) [inline modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/Main.java]: "application/vnd.wso2.governance-archive" Moved to a final string 
#367

File: modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/Main.java [hunk 5]
```diff
@@ -117,36 +141,94 @@ public void setCookie(String cookie) {
...
+     *This method is used to upload sample swagger docs using gar file.
+     *
+     * @param resourceServiceClient
+     * @param projectPath  absolute path of the gar file.
+     * @throws Exception
+     */
     private static void addSwaggerGar(ResourceServiceClient resourceServiceClient, String projectPath) throws Exception {
         DataHandler dh = new DataHandler(new URL("file://"+projectPath+"/resources/swagger.gar"));
         resourceServiceClient.addResource("/_system/governance/trunk/test/schemas/",
                 "application/vnd.wso2.governance-archive", null, dh, null,null);
```
Comment (Rajith90) [inline modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/Main.java]: "application/vnd.wso2.governance-archive" Moved to a final string 
#367

File: modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/RXTReDeploy.java [hunk 6]
```diff
@@ -25,12 +26,18 @@
...
 public class RXTReDeploy {
 
     private static String cookie;
     private static final String username = "admin";
     private static final String password = "admin";
-    private static final String serverURL = "https://localhost:9443/services/";
+    private static String port ;
+    private static String host ;
+    private static String serverURL;
     private static final String serviceRxtPath = "/_system/governance/repository/components/org.wso2.carbon.governance/types/";
```
Comment (Rajith90) [inline modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/RXTReDeploy.java]: Formatted in PR #367

File: modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/utils/SwaggerImportClient.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import org.apache.axis2.AxisFault;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.axis2.client.Options;
+import org.apache.axis2.client.ServiceClient;
+import org.apache.axis2.client.Stub;
+import org.wso2.carbon.registry.resource.stub.ResourceAdminServiceExceptionException;
+import org.wso2.carbon.registry.resource.stub.ResourceAdminServiceResourceServiceExceptionException;
+import org.wso2.carbon.registry.resource.stub.ResourceAdminServiceStub;
+import org.wso2.carbon.registry.resource.stub.beans.xsd.*;
```
Comment (Rajith90) [inline modules/samples/data-populator/Populator/src/org/wso2/carbon/registry/samples/populator/utils/SwaggerImportClient.java]: Used all bean classes.

**Pull request merged**